auction-server/src/auction/service/workers.rs

```diff
@@ -132,6 +137,53 @@ impl Service<Evm> {
 const GET_LATEST_BLOCKHASH_INTERVAL_SVM: Duration = Duration::from_secs(5);
 
 impl Service<Svm> {
+    pub async fn conclude_auction_for_log(
```

Comment:
remove `//TODO: this can be optimized out if triggered by websocket events` from auction_manager.rs since you implemented this :D